test/inductor/test_torchinductor_dynamic_shapes.py

```diff
@@ -277,6 +278,41 @@ def fn(a):
         actual = cfn(5)
         self.assertEqual(expect, actual)
 
+    def test_boolean_mask_rewritten(self, device):
+        counters.clear()
+
+        def fn(x, y):
+            x[y] *= 0.2
+            return x
```

Comment:
Are we maintaining the same mutation ? can you check that x and x1 are equal ?

Reply:
Yes, we only rewrite the when x and x1 are equal. We do this check in ```boolean_mask_rewritten```.